File: src/main/java/hilfling/backend/hilfling/controllers/PositionController.java
package hilfling.backend.hilfling.controllers;

import hilfling.backend.hilfling.model.Position;
import hilfling.backend.hilfling.service.PositionService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1/positions")
public class PositionController extends GenericBaseControllerImplementation<Position> {
    @Autowired
    private PositionService service;
    public PositionService getService() {
        return service;
    }
}

File: src/main/kotlin/hilfling/backend/hilfling/controller/PositionController.kt
package hilfling.backend.hilfling.controller

import hilfling.backend.hilfling.model.Album
import hilfling.backend.hilfling.model.Position
import hilfling.backend.hilfling.repository.AlbumRepository
import hilfling.backend.hilfling.repository.PositionRepository
import me.liuwj.ktorm.database.Database
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/positions")
class PositionController {
    val repository = PositionRepository()

    @GetMapping("/{id}")
    fun getById(@PathVariable("id") id: Int) : Position? {
        return repository.findById(id)
    }

    @GetMapping
    fun getAll() : List<Position> {
        return repository.findAll()
    }

    @PostMapping
    fun create(
            @RequestParam("position") position: Position
    ): Position {
        return repository.create(position)
    }
}